src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx

```diff
@@ -162,7 +162,7 @@ export class CreateDataSourceForm extends React.Component<
   };
 
   validateUsername = () => {
-    const isValid = !!this.state.auth.credentials.username?.trim().length;
+    const isValid = !!this.state.auth.credentials.username?.toString().trim().length;
```

Comment:
I guess we expect `username` to be a non-empty string? I'd do this validation instead:

```typescript
const isValid = typeof this.state.auth.credentials.username === 'string' && this.state.auth.credentials.username.trim().length > 0;
```